contract/src/main/kotlin/com/ritense/valtimo/contract/validation/Validatable.kt

```diff
@@ -24,17 +24,26 @@ import javax.validation.Validator
 
 interface Validatable {
 
-    val validator: Validator
-        @JsonIgnore get() = Validation.buildDefaultValidatorFactory().validator
-
     @JsonIgnore
     fun validate() {
         val logger = KotlinLogging.logger {}
         logger.debug { "validating $this" }
-        val errors = validator.validate(this)
+        val errors = getValidator().validate(this)
         if (errors.isNotEmpty()) {
             throw ConstraintViolationException(errors)
         }
     }
 
-}
\ No newline at end of file
+    companion object {
+        private var validator: Validator? = null
+
+        fun getValidator(): Validator {
+            return validator ?: Validation.buildDefaultValidatorFactory().validator
```

Comment:
Why not set it when it's null? You left a possibility open where it would still create a new validator every request.

Reply:
Fixed